sos/policies/runtimes/crio.py

```
@@ -0,0 +1,80 @@
+        containers = []
+        _cmd = "%s ps %s" % (self.binary, '-a' if get_all else '')
+        if self.active:
+            out = sos_get_command_output(_cmd, chroot=self.policy.sysroot)
+            if out['status'] == 0:
+                for ent in out['output'].splitlines()[1:]:
+                    ent = ent.split()
+                    # takes the form (container_id, container_name)
+                    containers.append((ent[0], ent[-3]))
+        return containers
+
+    def get_images(self):
+        """Get a list of images present on the system
+
+        :returns: A list of 2-tuples containing (image_name, image_id)
+        :rtype: ``list``
+        """
+        images = []
+        if self.active:
+            out = sos_get_command_output("%s images" % self.binary,
+                                         chroot=self.policy.sysroot)
+            if out['status'] == 0:
+                for ent in out['output'].splitlines():
+                    ent = ent.split()
+                    # takes the form (image_name, image_id)
+                    images.append((ent[0] + ':' + ent[1], ent[2]))
+        return images
+
+    def fmt_container_cmd(self, container, cmd, quotecmd):
+        """Format a command to run inside a container using the runtime
+
+        :param container: The name or ID of the container in which to run
+        :type container: ``str``
+
+        :param cmd: The command to run inside `container`
+        :type cmd: ``str``
+
+        :param quotecmd: Whether the cmd should be quoted.
+        :type quotecmd: ``bool``
+
+        :returns: Formatted string to run `cmd` inside `container`
+        :rtype: ``str``
+        """
+        if quotecmd:
+            quoted_cmd = quote(cmd)
+        else:
+            quoted_cmd = cmd
+        container_id = self.get_container_by_name(container)
+        return "%s %s %s" % (
+            self.run_cmd, container_id
+            if container_id is not None else container, quoted_cmd)
```

Why are we getting the ID here? Does crio not support the use of container names for exec?

---

I'm afraid it doesn't
```
USAGE:
   crictl exec [command options] CONTAINER-ID COMMAND [ARG...]
```
I'd like to return some kind of error here if container id is not found (command won't work anyway), but I found that general function https://github.com/sosreport/sos/blob/main/sos/report/plugins/__init__.py#L2423 just returns command if container is not found and followed the same rule here
Maybe you can help me figuring out what is the best option for error here?

---

For now, return an empty string. The changes (just merged) to `add_cmd_output()` will log an error.

Alternatively, if you want we could enclose the call to `fmt_container_cmd()` in `add_cmd_output()` in a `try` block, and you could raise an exception here in the crio runtime, to get catch by that block.

I'm fine either way.